### ISSUE TITLE
 transform OWL to TypeDB

### ISSUE THREAD

**Issue by wdduncan** [vaticle/typedb#6561]
I have a number of OWL ontologies that I work with. Do you have a tool for converting OWL to TypeDB? I can try making my own, but wanted to ask before doing so.

Sorry if this question has been asked elsewhere. My googling couldn't find the answer :(

**Comment by haikalpribadi** [comment]
I don't believe there is any, yet, @wdduncan . I think it would be a great idea if you'd like to build one and share it with the open source community (https://github.com/typedb-osi).

**Comment by thomaschristopherking** [comment]
Aren't there some fundamental differences between OWL/Description Logic and TypeDB? One is, being classical, the lack of weak negation in OWL. Conversely, I believe there is strong negation in OWL but not TypeDB? Although can represent it in TypeDB through other means.

Not sure if you want to include rules as well, but I think some OWL versions have more expressivity than TypeDB (not to take away from TypeDB, because the expressiveness of OWL isn't actually useful to people without experience in formal logic).

**Comment by haikalpribadi** [comment]
Actually, you have a good point there, @thomaschristopherking. You also reminded me of something more fundamental, that would make converting OWL ontologies **_"generically"_** to TypeQL impossible: OWL's data structure is based on subject-predicate-object, and TypeQL is entity-relation-attribute. Data models in OWL would be lower-level (and more granular) than they would be in TypeQL which has a higher-level representation. I.e. there are many ways to represent 1 TypeQL relationship model in OWL, and it's converting an OWL relationship model to TypeQL is ambiguous without human contextual input. It will be like converting assembly language to C/C++, which can't be done generically cause you write all kinds of assembly statements to do one thing in C/C++. However, the reverse is possible: TypeQL to OWL (just like C/C++ to assembly).

**Comment by wdduncan** [comment]
@haikalpribadi and @thomaschristopherking those are very good points! Some other considerations would be how to transform disjointness and quantifiers (i.e, `some`, `only`) into typedb.
